modules/network/src/main/java/org/apache/ignite/network/NetworkClusterContext.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network;
+
+import java.util.Map;
+
+/**
+ * Cluster context.
+ */
+public class NetworkClusterContext {
+    /** Message handlers. */
+    private final MessageHandlerHolder messageHandlerHolder;
+
+    /** Message mappers, message type -> message mapper. */
+    private final Map<Short, MessageMapper> messageMappers;
+
+    /**
+     * Constructor.
+     * @param messageHandlerHolder Message handlers.
+     * @param messageMappers Message mappers map.
+     */
+    public NetworkClusterContext(MessageHandlerHolder messageHandlerHolder, Map<Short, MessageMapper> messageMappers) {
+        this.messageHandlerHolder = messageHandlerHolder;
+        this.messageMappers = messageMappers;
+    }
+
+    /**
+     * @return Message handlers.
+     */
+    public MessageHandlerHolder messageHandlerHolder() {
+        return messageHandlerHolder;
+    }
+
+    /**
+     * @return Message mappers map.
+     */
+    public Map<Short, MessageMapper> messageMappers() {
+        return messageMappers;
```

Comment:
Should we wrap this in an `unmodifiableMap`?

Reply:
true